CRM/Contribute/BAO/ContributionRecur.php

```diff
@@ -821,13 +821,16 @@ public static function getRecurringFields() {
    *   Payment status - this correlates to the machine name of the contribution status ID ie
    *   - Completed
    *   - Failed
-   * @param string $effectiveDate
+   * @param string $contributionReceiveDate
    *
    * @throws \CiviCRM_API3_Exception
    */
-  public static function updateOnNewPayment($recurringContributionID, $paymentStatus, $effectiveDate) {
+  public static function updateOnNewPayment($recurringContributionID, $paymentStatus, $contributionReceiveDate) {
+    // If contributionReceiveDate matches next_sched_contribution_date we update next_sched_contribution_date by one period
+    $contributionReceiveDate = $contributionReceiveDate
+      ? date('Y-m-d H:i:s', strtotime($contributionReceiveDate))
+      : date('Y-m-d') . '00:00:00';
```

Comment:
Won't an extra space be required here?  Will it produce (for example) the string 2020-04-1600:00:00?